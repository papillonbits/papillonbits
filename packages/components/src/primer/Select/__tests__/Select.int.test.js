import { items, custom, regular, inactive } from './Select.int.story'

jest.mock('@papillonbits/library/array', () => ({
  getIndexItems: () => {},
  getIndexItemsWithSelected: () => {},
}))

const libraryArrayMockObject = require('@papillonbits/library/array')

describe('<Select />', () => {
  const indexItemsDataObject = [
    { ...items[0], index: 0 },
    { ...items[1], index: 1 },
    { ...items[2], index: 2 },
    { ...items[3], index: 3 },
  ]

  const newIndexItemsWithSelectedDataObject = [
    { ...items[0], index: 0, isSelected: true },
    { ...items[1], index: 1, isSelected: false },
    { ...items[2], index: 2, isSelected: false },
    { ...items[3], index: 3, isSelected: false },
  ]

  const newItemsDataObject = [
    { ...items[0], isSelected: true },
    { ...items[1], isSelected: false },
    { ...items[2], isSelected: false },
    { ...items[3], isSelected: false },
  ]

  const onChangeMockFn = jest.fn()

  beforeEach(() => {
    jest.spyOn(libraryArrayMockObject, 'getIndexItems').mockReturnValue(indexItemsDataObject)

    jest.spyOn(libraryArrayMockObject, 'getIndexItemsWithSelected').mockReturnValue(newIndexItemsWithSelectedDataObject)
  })

  afterEach(() => jest.clearAllMocks())

  describe('Event', () => {
    test('must return new items without index when onChange()', () => {
      const mountCustomRender = global.renderMount(custom(onChangeMockFn))
      mountCustomRender.find('select').first().simulate('change')
      expect(mountCustomRender.props().onChange).toBe(onChangeMockFn)
      expect(mountCustomRender.props().onChange).toHaveBeenCalledWith({
        selectedText: newItemsDataObject[0].text,
        items: newItemsDataObject,
      })
    })
  })

  describe('Render', () => {
    test('must match regular()', () => {
      expect(global.renderToJSON(regular())).toMatchSnapshot()
    })

    test('must match inacive()', () => {
      expect(global.renderToJSON(inactive())).toMatchSnapshot()
    })
  })
})
