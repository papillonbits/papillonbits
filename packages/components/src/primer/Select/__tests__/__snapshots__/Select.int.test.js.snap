// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Select /> Render must match inacive() 1`] = `
<div
  className="select"
>
  <select
    className="select-menu width-full select-inactive"
    id={null}
    onChange={[Function]}
    readOnly={false}
    value="Select"
  >
    <option
      selected={true}
      value="Select Item 1"
    >
      Select Item 1
    </option>
    <option
      selected={false}
      value="Select Item 2"
    >
      Select Item 2
    </option>
    <option
      selected={false}
      value="Select Item 3"
    >
      Select Item 3
    </option>
    <option
      selected={false}
      value="Select Item 4"
    >
      Select Item 4
    </option>
  </select>
</div>
`;

exports[`<Select /> Render must match regular() 1`] = `
<div
  className="select"
>
  <select
    className="select-menu width-full"
    id={null}
    onChange={[Function]}
    readOnly={false}
    value="Select"
  >
    <option
      selected={true}
      value="Select Item 1"
    >
      Select Item 1
    </option>
    <option
      selected={false}
      value="Select Item 2"
    >
      Select Item 2
    </option>
    <option
      selected={false}
      value="Select Item 3"
    >
      Select Item 3
    </option>
    <option
      selected={false}
      value="Select Item 4"
    >
      Select Item 4
    </option>
  </select>
</div>
`;
